sdk/clientcore/core/src/main/java/io/clientcore/core/http/pipeline/HttpInstrumentationPolicy.java

```diff
@@ -484,11 +487,12 @@ private static boolean canLogBody(BinaryData data) {
      * @param logBuilder Log message builder.
      */
     private void addHeadersToLogMessage(HttpHeaders headers, ClientLogger.LoggingEvent logBuilder) {
-        if (httpLogDetailLevel.shouldLogHeaders()) {
-            for (HttpHeader header : headers) {
-                HttpHeaderName headerName = header.getName();
-                String headerValue = allowedHeaderNames.contains(headerName) ? header.getValue() : REDACTED_PLACEHOLDER;
-                logBuilder.addKeyValue(headerName.toString(), headerValue);
+        for (HttpHeader header : headers) {
```

Comment:
Discussed briefly offline, I think we should go all the way with `HttpLoggingPolicy` and `HttpInstrumentationPolicy` being merged together as the more generically named `HttpInstrumentationPolicy`, which should support logging, metrics (when available), and tracing.

So, in this spot this will need to support the full capabilities of `HttpLoggingPolicy`.

Reply:
I have a small write-up in the PR description on this. Let me actually write down a gist to share the end-to-end story with other folks